### PR TITLE
use xmldoc instead of xml-js, with refactor to accomadate new package

### LINKED ISSUE #2363: Use a small XML parser
**Is your feature request related to a problem? Please describe.**
As described in this comment https://github.com/bbc/simorgh/pull/4184#issuecomment-540079051 the package `xml-js` in `psammead-rich-text-transforms` isn't big as a published size but the install size is significantly bigger than expected. 

**Describe the solution you'd like**
Find and use a smaller parser for `psammead-rich-text-transforms`

**Describe alternatives you've considered**
Accept the 90KB+ increase and split the bundles so `xml-js` is only included on page that require it to render. 

**Testing notes**
[Tester to complete]

Dev insight: Will there be any potential regression? etc

- ~[ ] This feature is expected to need manual testing.~

### Additional context
Tried and not suitable: 

**xml2json-light**
Ignores text between nodes 
`</paragraph><bold>this is okay</body>this is lost</paragraph>`

**fast-xml-parser**
Combines matching nodes and doesn't keep the order when multiple matched

Now trying
**xmldoc** 
Looks promising with the complicated example.

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #2363

**Overall change:** Swaps out `xml-js` to `xmldoc` as the dependency of this package and updates the code based on the new packages format.

**Code changes:**

- Swaps the checking the `node.type as text` to `node.constructor.name === 'XmlTextNode`
- Removes `findEl` and `findAtr` as the new format is less nested
- Has to loop over each child of the urlLink to build out the `text`, `locator` and `blocks`
- Swaps `elements` for `children` due to the new format 
- Checks if the array is empty with (`childNode.children.length !== 0`) before recursively calling `convertToBlocks()`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- ~[ ] This PR requires manual testing~

### CODE REVIEW

File: packages/utilities/psammead-rich-text-transforms/src/candy-xml.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ const attributeTags = ['bold', 'italic'];
 const supportedXmlNodeNames = ['paragraph', 'link', 'url', ...attributeTags];
 
 const isXmlNodeSupported = node => {
-  if (path(['type'], node) === 'text') {
-    return Boolean(pathOr('', ['text'], node).trim());
+  if (node.constructor.name === 'XmlTextNode') {
```
Comment (ryanmccombe) [inline packages/utilities/psammead-rich-text-transforms/src/candy-xml.js]:
```suggestion
return node.constructor.name === 'XmlTextNode';
```

**Review by ryanmccombe — APPROVED** [review]

**Review by ibMadbouly — APPROVED** [review]

File: packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js [hunk 2]
```diff
@@ -418,9 +413,9 @@ test('returns plain text if wrapped in an unsupport xml node', () => {
 
 test('can handle chaos', () => {
   const richText = candyXmlToRichText(
-    createBody(`
-      <paragraph><foo><bar><bold><meh>Bold text in unsupported nodes</meh></bold></bar></foo> followed by normal text then <italic><this><is>some <bold>carnage<carnage> </carnage></bold></is></this></italic>.</paragraph>
```
Comment (dr3) [inline packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: How come you removed the space in carnage? Was it because you had to?

Comment (pjlee11) [reply packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: I realised the previous logic didn't work with an empty XML tag `<carnage></carange>` so removing this space actually makes this test even more complicated

Comment (pjlee11) [reply packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: I'll add another explicit test for an empty XML tag

Comment (pjlee11) [reply packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: This pointed out a problem ... will push shortly

Comment (pjlee11) [reply packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: The fragment carnage gets lost in the current implementation...
image

😞

Comment (pjlee11) [reply packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js:422]: I don't think this should hold up this PR. I've created https://github.com/bbc/psammead/issues/2378 to address this, but I'm not convinced we will ever receive input this complicated so might not be required.

File: packages/utilities/psammead-rich-text-transforms/src/candy-xml.js (hunk 1 above)
Comment (dr3) [inline packages/utilities/psammead-rich-text-transforms/src/candy-xml.js]: should prob use `path` here

**Review by jroebu14 — APPROVED** [review]
